packages/dma-contracts/tasks/ownership-tool/index.ts

```diff
@@ -0,0 +1,301 @@
+    newOwnerAddress,
+  ])
+  const msgHash = ethers.utils.keccak256(msgData)
+
+  const requiredDelayBN = await system.serviceRegistry.requiredDelay()
+  const executionTimestampBN = await system.serviceRegistry.lastExecuted(msgHash)
+
+  const now = (await ethers.provider.getBlock('latest')).timestamp
+
+  const requiredDelay = requiredDelayBN.toNumber()
+  const requestTimestamp = executionTimestampBN.toNumber()
+
+  const isPending = requestTimestamp != 0
+  const remainingTime = requestTimestamp + requiredDelay - now
+
+  return {
+    serviceRegistry: {
+      isPending,
+      waitingTimeSecs: remainingTime < 0 ? 0 : remainingTime,
+    },
+  }
+}
+
+async function changeOwners(newOwnerAddress: string, network: string, ethers: any) {
+  const signer = (await ethers.getSigners())[0]
+
+  if (isInvalidAddress(newOwnerAddress)) {
+    throw new Error(`Invalid address for new owner: ${newOwnerAddress}`)
+  }
+
+  const system = await getSystem(network, ethers, signer)
+  if (!system) {
+    return
+  }
+
+  const systemOwnership = await getSystemOwnership(system)
+  if (!systemOwnership) {
+    return
+  }
+
+  const isServiceRegistryChangeNeeded = systemOwnership.serviceRegistry.owner !== newOwnerAddress
+  const isOperationsRegistryChangeNeeded =
+    systemOwnership.operationsRegistry.owner !== newOwnerAddress
+
+  if (!isServiceRegistryChangeNeeded) {
+    console.log(`The new owner address is already the owner of the ServiceRegistry`)
+  }
+  if (!isOperationsRegistryChangeNeeded) {
+    console.log(`The new owner address is already the owner of the OperationsRegistry`)
+  }
+
+  if (!isServiceRegistryChangeNeeded && !isOperationsRegistryChangeNeeded) {
+    console.log('No changes needed')
+    return
+  }
+
+  console.log('\n============================')
+  console.log('====== CHANGING OWNER ======')
+  console.log('============================\n')
+
+  printOwnership(systemOwnership, network, 'CURRENT CONFIG')
+
+  console.log(`======= NEW OWNER DATA =======`)
+  console.log(`Signer address: ${signer.address}`)
+  console.log(`New owner address: ${newOwnerAddress}`)
+  console.log('==============================\n')
+
+  console.log(`!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!`)
+  console.log(`!!!!!!!!!        WARNING: THIS IS A DANGEROUS OPERATION          !!!!!!!!!`)
+  console.log(`!!!!!!!!! TAKE YOUR TIME TO VERIFY THAT THE NEW OWNER IS CORRECT !!!!!!!!!`)
+  console.log(`!!!!!!!!! OTHERWISE THE CONTRACTS OWNERSHIP WILL BE LOST FOREVER !!!!!!!!!`)
+  console.log(`!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n`)
+  const answer = await input({
+    message: `Check the config above first. You are about to change the owner to ${newOwnerAddress}'. Do you want to continue? (yes/NO)`,
+  })
+
+  if (answer === 'y' || answer === 'Y') {
+    console.log(
+      "Please write 'yes' with all the letters to confirm the operation...aborting now...",
```

Comment:
why "...aborting now..." in this message?

Reply:
I didn't want to have a loop there to retry. It is a dangerous operation, so if you didn't know that you had to write `yes` I prefer to abort so you have to re-run it